### PR TITLE
HPCC-17191 ecl result command to output full workunit results

### PR DESCRIPTION
**Pull request by afishbeck**
Signed-off-by: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [x] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by afishbeck** [comment]
@wangkx if you have time, can you review this as well?

File: ecl/eclcmd/eclcmd_core.cpp [hunk 1]
```diff
@@ -939,6 +939,164 @@ class EclCmdRun : public EclCmdWithEclTarget
...
+        req->setIncludeXmlSchemas(false);
+        req->setIncludeResourceURLs(false);
+        req->setSuppressResultSchemas(true);
+        Owned<IClientWUInfoResponse> resp = client->WUInfo(req);
+        int ret = outputMultiExceptionsEx(resp->getExceptions());
+
+        IConstECLWorkunit &wu = resp->getWorkunit();
+        IArrayOf<IConstECLException> &exceptions = wu.getExceptions();
+        unsigned count = wu.getResultCount();
+        if (count<=0 && exceptions.ordinality()<=0)
```
Comment (wangkx) [inline ecl/eclcmd/eclcmd_core.cpp]: Do you need to check the ret here?

**Review by wangkx — COMMENTED** [review]
@afishbeck just 1 question.

File: ecl/eclcmd/eclcmd_core.cpp (hunk 1 above)
Comment (afishbeck) [inline ecl/eclcmd/eclcmd_core.cpp]: Checking ret or exceptions.ordinality() are equivalent here.  But it may be clearer to use ret.  I'll change it.

**Comment by afishbeck** [comment]
@wangkx decided that returning an empty `<Result>` was better.  Please check.

**Comment by wangkx** [comment]
@afishbeck looks good.

**Pull request merged**